GA Install Astro on AWS Using a Cross-account Role (#1958)
astronomer/docs: astro/install-aws.md

- Create an account on Astro.
- Share AWS account information with Astronomer support.

> - Create the IAM policies used by Astro. This includes a cross-account IAM role that Astro can assume and permissions boundaries for the Astro operational roles. For more information about permissions boundaries, see [Permissions boundaries for IAM entities](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html).

Reviewer: ```suggestion
- Create the IAM policies used by Astro. This includes a cross-account IAM role that Astro can assume and [permissions boundaries](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) for the Astro operational roles.
```
For v optional reference content, I prefer if we save real estate by hyperlinking to the word being defined. Don't feel strongly either way tho.

    - `https://updates.astronomer.io/`
    - `https://install.astronomer.io/`

> Astro requires a clean AWS account with the minimum EC2 service quotas. For security reasons, the install process is not currently supported on an AWS account that has other tooling running in it. For instructions on creating a new AWS account, see [How do I create and activate a new AWS account?](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/).

Reviewer 2: ```suggestion
Astro requires a dedicated AWS account with the minimum EC2 service quotas. For security reasons, the install process is not currently supported on an AWS account that has other resources running in it. For instructions on creating a new AWS account, see [How do I create and activate a new AWS account?](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/).
```

| QuotaCode  | QuotaName                                                        | Minimum Value  |
| -----------| ---------------------------------------------------------------- | ---------------|
| L-1216C47A | Running On-Demand Standard (A, C, D, H, I, M, R, T, Z) instances | 40             |
| L-34B43A08 | All Standard (A, C, D, H, I, M, R, T, Z) Spot Instance Requests  | 40             |

These quotas are required to mitigate near term capacity risks and simplify the Astro onboarding experience. If you need to modify or increase a specific quota, see [Request a quota increase](https://docs.aws.amazon.com/servicequotas/latest/userguide/request-quota-increase.html).

:::tip

If you have one or more existing AWS accounts, you can use [AWS Organizations](https://aws.amazon.com/organizations/) to manage billing, users, and more in a central place. For more information on how to add your Astro AWS account to your AWS Organization, see [Inviting an AWS account to join your organization](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_accounts_invites.html).

:::

#### VPC peering prerequisites (Optional)

> The following options are available when your AWS resources are on a private network:

Reviewer: ```suggestion
The following options are available to connect Astro to AWS resources on a private network:
```

> - Allow traffic through the public internet and use allow-lists for communication.

Reviewer: ```suggestion
- Allow traffic through the public internet and use allowlists for communication.
```

### Retrieve an external ID from the Cloud UI

You must be an Organization Owner to view the external ID. If you are not an Organization Owner, the **AWS External ID** field will not appear in the Cloud UI.

1. In the Cloud UI, click the **Settings** tab.

2. Click **Show** in the **AWS External ID** field and then click **Copy**. This external ID is a unique identifier that Astro uses to connect to your AWS account. 

3. Save the external ID as a secret or in another secure format. See [How to use an external ID when granting access to your AWS resources to a third party](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).

### Create a cross-account role

Use the external ID to create a cross-account IAM role for Astro.

> 1. Log in to your AWS CloudFormation instance.

Reviewer 2: ```suggestion
1. Log in to your [AWS CloudFormation console](https://console.aws.amazon.com/cloudformation).
```

2. Open the [Astronomer cross-account role CloudFormation template](https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review?templateURL=https://astro-cross-account-role-template.s3.us-east-2.amazonaws.com/astronomer-remote-management-stack.yaml&stackName=AstroCrossAccountRole).

3. Enter the external ID that you copied in Step 2 in the **ExternalId** field.

4. Select the **I acknowledge that AWS CloudFormation might create IAM resources with custom names** checkbox.

5. Click **Create Stack**.

#### Cross-account role modifications

> When new features or functionality are added to Astro, Astronomer might need to modify cross-account role permissions. When setting permissions, Astronomer adheres to the least-privilege permissions standard and adds only the permissions necessary to use the new feature or functionality.

Reviewer: ```suggestion
When new features or functionality are added to Astro, Astronomer might need to modify cross-account role permissions. When setting permissions, Astronomer adheres to the least-privilege permissions standard and adds only the permissions necessary for new features or functionality.
```

> Astronomer support notifies your Organization when any changes are made to the policies that expand user access. Notifications will include an explanation of the changes being made and why the change was necessary. 

Reviewer 2: ```suggestion
Astronomer support notifies your Organization when any changes are made to the policies that expand cross-account role access. Notifications will include an explanation of the changes being made and why the change was necessary. 
```

Astronomer can reduce the access available to the policies without notification.

#### Monitor the policies for changes (optional)

> You can use CloudTrail to monitor changes to Astro policies. Access to CloudTrail has been limited to prevent the accidental modification or deletion of CloudTrail logs by Astronomer support. The following table lists the events that you should monitor. 

Reviewer: ```suggestion
You can use CloudTrail to monitor changes to Astro policies. The following table lists the events that you should monitor. 
```
Not sure what this sentence is doing here, let's remove it and I'll add it back once I figure out when it was added and why?

Reviewer 2: Let's leave it. This is actually an important detail.
